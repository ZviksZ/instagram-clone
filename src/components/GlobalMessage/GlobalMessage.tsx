import React, {useEffect}         from 'react';
import {useDispatch, useSelector} from "react-redux";
import {AppState}                 from "../../redux/store";
import MuiAlert, { AlertProps }   from '@material-ui/lab/Alert';
import {setMessageData}           from "../../redux/app-reducer";
import s                          from './GlobalMessage.module.scss'

function Alert(props: AlertProps) {
   return <MuiAlert className={s.alert} elevation={6} variant="filled" {...props} />;
}

export const GlobalMessage = () => {
   const message = useSelector((state: AppState) => state.app.message);
   const dispatch =useDispatch();

   useEffect(() => {
      if (message !== null) {
         setTimeout(() => {
            dispatch(setMessageData(null))
         }, 3000)      }
   }, [message])
   return (
      <>
         {
            message && <Alert severity={message.type}>{message.text}</Alert>
         }
      </>
   );
}

