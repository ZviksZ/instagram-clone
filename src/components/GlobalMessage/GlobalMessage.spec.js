import { useSelector } from 'react-redux';
import React                         from "react";
import {GlobalMessage}               from "./GlobalMessage";

jest.mock("react-redux", () => ({
   useDispatch: jest.fn(),
   useSelector: jest.fn(),
}));


const setUp = () => shallow(<GlobalMessage />);

describe("GlobalMessage component", () => {
   afterEach(() => {
      jest.clearAllMocks();
   });

   afterAll(() => {
      jest.restoreAllMocks();
   });

   it("should render GlobalMessage", () => {
      useSelector.mockImplementation((selector) => selector({
         app: {
            message: {
               type: 'error',
               text: 'Ошибка. Попробуйте снова'
            }
         }
      }));

      const component = shallow(<GlobalMessage />);
      expect(component).toMatchSnapshot();
   });
});
