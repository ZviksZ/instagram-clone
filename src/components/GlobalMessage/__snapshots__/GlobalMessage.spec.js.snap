// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalMessage component should render GlobalMessage 1`] = `
<Fragment>
  <Alert
    severity="error"
  >
    Ошибка. Попробуйте снова
  </Alert>
</Fragment>
`;
